docs/C_API.md

```diff
@@ -21,13 +21,22 @@ is as follows
    * Create env using ```CreateEnvWithGlobalThreadPools()```
    * Create session and call ```DisablePerSessionThreads()``` on the session options object
    * Call ```Run()``` as usual
-* **Share allocator(s) between sessions:** Allow multiple sessions in the same process to use the same allocator(s). This
-allocator is first registered in the env and then reused by all sessions that use the same env instance unless a session
-chooses to override this by setting ```session_state.use_env_allocators``` to "0". Usage of this feature is as follows
-   * Register an allocator created by ORT using the ```CreateAndRegisterAllocator``` API.
-   * Set ```session.use_env_allocators``` to "1" for each session that wants to use the env registered allocators.
-   * See test ```TestSharedAllocatorUsingCreateAndRegisterAllocator``` in
+* **Share allocator(s) between sessions:**
+   * *Description*: This feature allows multiple sessions in the same process to use the same allocator(s).
+   * *Scenario*: You've several sessions in the same process and see high memory usage. One of the reasons for this is as follows. Each session creates its own CPU allocator which is arena based by default. [ORT implements](onnxruntime/core/framework/bfc_arena.h) a simplified version of an arena allocator that is based on [Doug Lea's best-first with coalescing algorithm](http://gee.cs.oswego.edu/dl/html/malloc.html). This allocator allocates a large region of memory during init time and threafter it chunks, coalesces and extends this initial region as per allocation/deallocation demands. Since each allocator lives in its own session, there is no knowledge of memory allocations/deallocations across sessions. This leads to fragmentation and increased memory usage.
```

Comment:
threafter -> thereafter

Should you also mention that the arena never shrinks?